### PR TITLE
[EICNET-1665] Share group contents between groups

### PR DESCRIPTION
**Pull request by korhankale**

### CODE REVIEW

**Review by Askerx9 — APPROVED** [review]

File: lib/modules/eic_groups/modules/eic_share_content/src/Controller/ShareContentController.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+   * @var \Drupal\eic_share_content\Service\ShareManager
+   */
+  private $shareManager;
+
+  /**
+   * @var \Symfony\Component\HttpFoundation\Request|null
+   */
+  private $currentRequest;
+
+  public function __construct(
```
Comment (Dromadoros) [inline lib/modules/eic_groups/modules/eic_share_content/src/Controller/ShareContentController.php:32]: docblock

File: lib/themes/eic_community/react/components/Block/Modal/index.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import React, {useState} from 'react'
```
Comment (Dromadoros) [inline lib/themes/eic_community/react/components/Block/Modal/index.js:1]: I think the name is not correct, the "modal" name is too generic for what it does really, it's more a "share content modal"

File: lib/themes/eic_community/react/components/Block/Modal/index.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                  <div className="ecl-select__icon" dangerouslySetInnerHTML={{__html: svg("arrow-down", "ecl-icon ecl-icon--s ecl-icon--rotate-180 ecl-select__icon-shape")}} />
+                </div>
+              </div>
+
+              <div className="ecl-form-group">
+                <label className="ecl-form-label">
+                  Message
+                  <span className="ecl-form-label__required">*</span>
+                </label>
+                <div className="ecl-help-block">Your description can contain up to 300 characters.</div>
```
Comment (Dromadoros) [inline lib/themes/eic_community/react/components/Block/Modal/index.js:70]: We should send translations also in props

File: lib/themes/eic_community/templates/blocks/eic-share-content-link.html.twig [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{{ attach_library('eic_community/react-block-modal') }}
+<div id="react-modal-block" data-groups="{{ groups | json_encode }}" data-endpoint="{{ endpoint }}">
```
Comment (Dromadoros) [inline lib/themes/eic_community/templates/blocks/eic-share-content-link.html.twig:2]:
```suggestion
<div id="react-modal-block" data-groups="{{ groups | json_encode }}" data-endpoint="{{ endpoint }}" />
```

**Comment by Dromadoros** [comment]
It's wokring but it missed some front-end stuff like a loader when you click on share button, a success message or something saying that the content has been shared correctly

also an icon on the link is missing

**Pull request merged**